### PR TITLE
[Spark-7475][MLlib] adjust ldaExample for online LDA

### PR DESCRIPTION
**Pull request by hhbyyh**
jira: https://issues.apache.org/jira/browse/SPARK-7475

Add a new argument to specify the algorithm applied to LDA, to exhibit the basic usage of LDAOptimizer.

cc @jkbradley

### CODE REVIEW

File: examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala [hunk 1]
```diff
@@ -78,6 +79,10 @@ object LDAExample {
         .text(s"filepath for a list of stopwords. Note: This must fit on a single machine." +
         s"  default: ${defaultParams.stopwordFile}")
         .action((x, c) => c.copy(stopwordFile = x))
+      opt[String]("algorithm")
+        .text(s"concrete inference algorithm to use. em and online are supported." +
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala]: Remove word "concrete" (not needed)

File: examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala [hunk 2]
```diff
@@ -128,7 +133,16 @@ object LDAExample {
 
     // Run LDA.
     val lda = new LDA()
-    lda.setK(params.k)
+
+    val optimizer = params.algorithm match {
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala]: Convert algorithm to lower case to be a little more flexible

File: examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala [hunk 3]
```diff
@@ -128,7 +133,16 @@ object LDAExample {
...
+      case "em" => new EMLDAOptimizer
+      case "online" => new OnlineLDAOptimizer().setMiniBatchFraction(0.05 + 1.0 / actualCorpusSize)
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala]: Please add a small comment saying the adjustment is to be more robust on tiny datasets.

File: examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala [hunk 4]
```diff
@@ -137,14 +151,18 @@ object LDAExample {
...
+    val ldaModel = lda.run(corpus)
     val elapsed = (System.nanoTime() - startTime) / 1e9
 
     println(s"Finished training LDA model.  Summary:")
     println(s"\t Training time: $elapsed sec")
-    val avgLogLikelihood = ldaModel.logLikelihood / actualCorpusSize.toDouble
-    println(s"\t Training data average log likelihood: $avgLogLikelihood")
-    println()
+
+    if(ldaModel.isInstanceOf[DistributedLDAModel]) {
```
Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala]: scala style: space after "if"

**Comment by jkbradley** [comment]
That's all, I believe.  (I assume you've tested it?)  Thanks!

**Comment by jkbradley** [comment]
PS: Nice going hitting PR number 6000!

**Comment by hhbyyh** [comment]
Thanks for review. 
About 6000, there should be some kind of lottery for PR of X * 1000 ~~

**Comment by jkbradley** [comment]
@hhbyyh LGTM  thanks!  Merging into master and branch-1.4

**Comment by redsofa** [comment]
A slightly related question... Is there a way to get at the intermediate topic lists for each mini-batch ?

**Comment by hhbyyh** [comment]
@redsofa I would advise 
copy the MLlib LDAOptimizer code to your own project, 
add related logging to next() and just run it with your application code.

**Pull request closed without merging**